pulp_smash/tests/platform/api_v2/test_repository.py

```diff
@@ -0,0 +1,226 @@
+        for i, response in enumerate(self.responses):
+            with self.subTest(self.bodies[i]):
+                self.assertEqual(
+                    self.url + self.bodies[i]['id'] + '/',
+                    response.headers['Location']
+                )
+
+    def test_attributes(self):
+        """Assert that each repository has the requested attributes."""
+        for i, body in enumerate(self.bodies):
+            with self.subTest(body):
+                attributes = self.responses[i].json()
+                self.assertLessEqual(set(body.keys()), set(attributes.keys()))
+                attributes = {key: attributes[key] for key in body.keys()}
+                self.assertEqual(body, attributes)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the created repositories."""
+        for response in cls.responses:
+            requests.delete(
+                cls.cfg.base_url + response.json()['_href'],
+                **cls.cfg.get_requests_kwargs()
+            ).raise_for_status()
+
+
+class CreateFailureTestCase(TestCase):
+    """Establish that repositories are not created in documented scenarios."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create several repositories.
+
+        Create one repository with the minimum required attributes, and a
+        second with all available attributes except importers and distributors.
+
+        """
+        cls.cfg = get_config()
+        cls.url = cls.cfg.base_url + REPOSITORY_PATH
+        identical_id = rand_str()
+        cls.bodies = (
+            (201, {'id': identical_id}),
+            (400, {'id': None}),
+            (400, ['Incorrect data type']),
+            (400, {'missing_required_keys': 'id'}),
+            (409, {'id': identical_id}),
+        )
+        cls.responses = tuple((
+            requests.post(
+                cls.url,
+                json=body[1],
+                **cls.cfg.get_requests_kwargs()
+            )
+            for body in cls.bodies
+        ))
+
+    def test_status_code(self):
+        """Assert that each response has the expected HTTP status code."""
+        for i, response in enumerate(self.responses):
+            with self.subTest(self.bodies[i]):
+                self.assertEqual(response.status_code, self.bodies[i][0])
+
+    def test_location_header(self):
+        """Assert that the Location header is correctly set in the response."""
+        for i, response in enumerate(self.responses):
+            with self.subTest(self.bodies[i]):
+                if self.bodies[i][0] == 201:
+                    self.assertEqual(
+                        self.url + self.bodies[i][1]['id'] + '/',
+                        response.headers['Location']
+                    )
+                else:
+                    self.assertNotIn('Location', response.headers)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the created repositories."""
```

Comment:
Each test case has identical `tearDownClass` methods, except for this test case. I'd suggest defining a parent class or mixin class like so:

```python
def _BaseTestCase(TestCase):
    @classmethod
    def tearDownClass(cls):
        for response in cls.responses:
            requests.delete(...).raise_for_status()
```

And you can make use of it here like so:

```python
@classmethod
def tearDownClass(cls):
    cls.responses = (cls.responses[0],)
```

All that said, working with response objects directly like this is a bit of a kludge, as is the use of numeric indices. So you may wish to leave things as is and later refactor things as per the example in #26.